clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/security/CloudFoundryCredentialsSynchronizer.java

```diff
@@ -90,12 +80,10 @@ private void synchronizeRepository(List<CloudFoundryConfigurationProperties.Mana
     });
   }
 
-  private void synchronizeProvider(List<Agent> addedAgents) {
-    cloudFoundryProvider.getAgents().addAll(addedAgents);
-  }
-
   private void synchronizeAgentCache(List<Agent> addedAgents, List<String> deletedAccountNames) {
+    cloudFoundryProvider.getAgents().addAll(addedAgents);
```

Comment:
I'm not too familiar with the Cloud Foundry provider, or with the scheduling of caching agents in general, so this could use a look from someone else.  But if I'm reading this properly (which I may not be), it looks like if an account changes we'll store the new account details in the repository but we won't create a new caching agent and will leave the old caching agent running with a reference to the old `CloudFoundryCredentials`.  We probably don't want to reschedule all agents on every polling cycle, so it might be necessary to do some diffing of the existing and new account if we want to support live modifications.

Reply:
Unschedling and rescheduling accounts was added in another commit.